api/src/main/java/de/valtech/aecu/api/service/AecuService.java

```diff
@@ -137,6 +147,15 @@ public interface AecuService {
      */
     List<HistoryEntry> getHistory(int startIndex, int count) throws AecuException;
 
+    /**
+     * Executes the script(s) at the given position and taking install hook history into account.
```

Comment:
the javadoc is wrong as we cannot pass in the position. also in the implementation we have /apps hardcoded but scripts could also be stored elsewhere, e.g. within /conf . 

Suggestion: We remove the executeAllScripts method and implement the JMX interface for method `execute(String path, String data)`

Reply:
> the javadoc is wrong as we cannot pass in the position. also in the implementation we have /apps hardcoded but scripts could also be stored elsewhere, e.g. within /conf .
> 
> Suggestion: We remove the executeAllScripts method and implement the JMX interface for method `execute(String path, String data)`

I get that, but that is not the point of this method. The execute(String path, String data) is only for 1 groovyscript, while  the execute all scripts is for all script. 

Counter suggestion: executeScripts(String[] scriptRoots, String data)

What do you think?

Reply:
> > the javadoc is wrong as we cannot pass in the position. also in the implementation we have /apps hardcoded but scripts could also be stored elsewhere, e.g. within /conf .
> > Suggestion: We remove the executeAllScripts method and implement the JMX interface for method `execute(String path, String data)`
> 
> I get that, but that is not the point of this method. The execute(String path, String data) is only for 1 groovyscript, while the execute all scripts is for all script.
> 
> Counter suggestion: executeScripts(String[] scriptRoots, String data)
> 
> What do you think?

Hi @DriesVanbilloen 
Mmh, I just tested and it is enough to pass in a root path 
![grafik](https://user-images.githubusercontent.com/46745410/220285068-f836b853-052b-4af2-93c7-bd9a23615325.png)
Internally findCandidates is used https://github.com/valtech/aem-easy-content-upgrade/blob/master/core/src/main/java/de/valtech/aecu/core/service/AecuServiceImpl.java#L111

Reply:
Mmh, missed that. Ok I will apply your suggestions